hyperspy/learn/mva.py

```diff
@@ -1013,13 +1015,30 @@ def plot_explained_variance_ratio(self, n=None, log=True, threshold=0,
         # Determine right number of components for signal and cutoff value
         if isinstance(threshold, float):
```

Comment:
I would also check if `0 < t < 1`. If it is not, then it should raise a `ValueError`.